tcms/rpc/api/testcase.py

```diff
@@ -279,8 +283,24 @@ def filter(query=None):  # pylint: disable=redefined-builtin
     if query is None:
         query = {}
 
-    return list(
-        TestCase.objects.filter(**query)
+    # Cannot use `setup_duration` and `testing_duration` as aliases because Django
+    # doesn't allow annotating querysets with aliases which match existing fieldnames.
+    # https://code.djangoproject.com/ticket/11256
+    # https://github.com/django/django/blob/main/django/db/models/query.py#L1154-L1157
+
+    # we're adding the suffix `_ms` to the aliases because they are
+    # `Cast`ed to their millisecond values
+    qs = (
+        TestCase.objects.annotate(
+            expected_duration_ms=Cast(
+                Coalesce("setup_duration", Cast(timedelta(0), DurationField()))
+                + Coalesce("testing_duration", Cast(timedelta(0), DurationField())),
+                FloatField(),
+            ),
```

Comment:
Simpler -> `expected_duration_sec=(F('testing_duration_sec') + F('setup_duration_sec'))`. Also seems to return None for me when fields are none.


also swap the order of the fields for better readability.

Reply:
I've just tried this on my local database, and `expected_duration_sec=(F('testing_duration_sec') + F('setup_duration_sec'))` does not work correctly when only one of the fields is `None` -- `expected_duration_sec` becomes `None` regardless of the value of the other field.

I've added a test that exposes this to my current solution.

Reply:
Good catch, however I think we still need some changes here. 

If any of the setup_duration/testing_duration fields isn't set then the query should still return None as the value. However for expected_duration_sec always default to a numeric value to mimic how TestCase.expected_duration works.

I think if you move the Coalesce call only in the definition of expected_duration_sec that would solve the issue and also simplify the entire expression.